### PR TITLE
Add auto hosting plugin for internal use

### PR DESCRIPTION
**Pull request by tuliren**
This PR revives #21.

The auto hosting plugin is useful internally to help us figure out what build artifacts can be hosted in any repo. See https://github.com/stoat-dev/stoat/issues/175 for the specific reason.

It won't be used by default here. And we don't need to document it.

Resolve https://github.com/stoat-dev/stoat/issues/175.

### CODE REVIEW

File: .stoat/config.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ enabled: true
 plugins:
   job_runtime:
     enabled: true
+  auto_hosting:
+    enabled: true
```
Comment (tuliren) [inline .stoat/config.yaml:8]: Will remove this before merging.

**Review by jrhizor — APPROVED** [review]
Seems alright as-is, but maybe it would be better to always log the related paths instead of actually trying to host all of them? Or maybe having it as part of `stoat init` and prompt to actually enable each?

File: action/src/plugins/autoHosting/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  );
+  const rootDirectories = getRootDirectories(allDirectories);
+  core.info(`[${taskId}] Candidate directories:\n-- ${rootDirectories.join('\n-- ')}`);
+
+  const validDirectories = getValidDirectories(rootDirectories);
+
+  for (const directory of validDirectories) {
+    const staticHostingTaskConfig: StaticHostingPlugin = {
+      metadata: {
+        name: `Path \`${directory}\``
```
Comment (jrhizor) [inline action/src/plugins/autoHosting/plugin.ts:79]:
```suggestion
        name: `\`${directory}\``
```

**Comment by tuliren** [comment]
Right. I thought about this too. I think this plugin can be renamed to something like "html_suggestion", and the detected paths are rendered in a suggestion section in the comment at the bottom. Artifacts are only uploaded if a "auto_upload" config is set to "true".

The stoat init approach may not work because people don't typically run all their ci locally. So many of the artifacts won't exist locally.

**Comment by tuliren** [comment]
Should also exclude the already hosted paths.

**Comment by tuliren** [comment]
Maybe auto hosting is a legit use case. For example, airbyte can just use auto hosting for all the test reports. In this way, there is no need to manually list out all of them, even better than glob pattern. Users can define a hosting ignore file to exclude certain files and avoid secret leckage. Also personally, I don't want to figure out the path myself at all after doing this for a few repos.

One way to do this is to make the suggestion or auto hosting a special task under static hosting:

```yaml
static_hosting:
  task1:
    path: coverage/lcov
  # either this
  suggestion:
    html:
      enable: true
      auto_upload: true
  # or this
  auto_hosting:
    enable: true
    exclude: |
      .env*
      secret*
```

**Comment by tuliren** [comment]
@jrhizor, I have updated the auto hosting plugin as follows:
- It is activate by default, and will log any candidate paths for hosting in the Stoat action.
- It can be turned off by `enabled: false`.
- It will not auto upload the artifacts, unless `auto_upload: true`.

In the future, we should flush out the suggested paths for hosting in the comment. The config field is added to the schema, but it is not implemented yet.

**Pull request merged**